src/main/kotlin/org/jitsi/jibri/selenium/JibriSelenium.kt

```diff
@@ -0,0 +1,161 @@
+ * Selenium.  It:
+ * 1) Owns the webdriver
+ * 2) Handles passing the proper options to Chrome
+ * 3) Sets the necessary localstorage variables before joining a call
+ * It implements [StatusPublisher] to publish its status
+ */
+class JibriSelenium(
+    private val jibriSeleniumOptions: JibriSeleniumOptions,
+    private val executor: ScheduledExecutorService
+) : StatusPublisher<JibriServiceStatus>() {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+    private var chromeDriver: ChromeDriver
+    private val baseUrl: String = jibriSeleniumOptions.callParams.callUrlInfo.baseUrl
+    /**
+     * The options we'll add as url params
+     */
+    private val urlOptions = listOf(
+        "config.iAmRecorder=true",
+        "config.externalConnectUrl=null",
+        "interfaceConfig.APP_NAME=\"Jibri\""
+    )
+    /**
+     * A task which checks if Jibri is alone in the call
+     */
+    private var emptyCallTask: ScheduledFuture<*>? = null
+
+    /**
+     * Set up default chrome driver options (using fake device, etc.)
+      */
+    init {
+        System.setProperty("webdriver.chrome.logfile", "/tmp/chromedriver.log")
+        val chromeOptions = ChromeOptions()
+        chromeOptions.addArguments(
+                "--use-fake-ui-for-media-stream",
+                "--start-maximized",
+                "--kiosk",
+                "--enabled",
+                "--enable-logging",
+                "--vmodule=*=3",
+                "--disable-infobars",
+                "--alsa-output-device=plug:amix"
+        )
+        val chromeDriverService = ChromeDriverService.Builder().withEnvironment(
+            mapOf("DISPLAY" to jibriSeleniumOptions.display)
+        ).build()
+        chromeDriver = ChromeDriver(chromeDriverService, chromeOptions)
+    }
+
+    /**
+     * Set various values to be put in local storage.  NOTE: the driver
+     * should have already navigated to the desired page
+     */
+    private fun setLocalStorageValues(vararg keyValues: Pair<String, String>) {
+        for ((key, value) in keyValues) {
+            chromeDriver.executeScript("window.localStorage.setItem('$key', '$value')")
+        }
+    }
+
+    /**
+     * Check if Jibri is the only participant in the call.  If it is the only
+     * participant for 30 seconds, it will leave the call.
+     */
+    private fun addEmptyCallDetector() {
+        var numTimesEmpty = 0
+        emptyCallTask = executor.scheduleAtFixedRate(period = 15, unit = TimeUnit.SECONDS) {
+            try {
+                // >1 since the count will include jibri itself
+                if (CallPage(chromeDriver).getNumParticipants(chromeDriver) > 1) {
+                    numTimesEmpty = 0
+                } else {
+                    numTimesEmpty++
+                }
+                if (numTimesEmpty >= 2) {
+                    logger.info("Jibri has been in a lonely call for 30 seconds, marking as finished")
```

Comment:
Since the task was submitted without a delay, I think this will actually trigger after 15 seconds, not 30.